src/tag_gc.rs

```diff
@@ -6,6 +6,7 @@ use crate::*;
 
 pub trait VisitTags {
     fn visit_tags(&self, visit: &mut dyn FnMut(BorTag));
+    fn visit_prov(&self, visit: &mut dyn FnMut(AllocId));
```

Comment:
So now we have to copy-paste the traversal logic in every `impl VisitTags`, that's not great.

The alternative would be to make this `visit_provenance(&self, visit: &mut dyn FnMut(Provenance))`. But then the problem is that the SB stacks only have a `BorTag`, not an `AllocId`... and possibly somewhere we have an `AllocId` and not a `BorTag`? So what about
```
visit_provenance(&self, visit: &mut dyn FnMut(Option<AllocId>, Option<BorTag>))
```

Reply:
Originally I didn't want to do this because I thought we would have reason to run the AllocId collection much less often than the BorTag collection, but I think AllocIds are just so much more rare that it's not worth doing such an optimization.